app/scripts/streams/provider-stream.ts

```diff
@@ -226,19 +226,33 @@ const onMessageSetUpExtensionStreams = (msg: MessageType) => {
   return undefined;
 };
 
+/**
+ * Ends two-way communication streams between browser extension and
+ * the local per-page browser context.
+ */
+export function destroyStreams() {
+  if (!extensionPort) {
+    return;
+  }
+  extensionPort.onDisconnect.removeListener(onDisconnectDestroyStreams);
+
+  destroyExtensionStreams();
+  destroyLegacyExtensionStreams();
+
+  extensionPort.disconnect();
```

Comment:
Should `METAMASK_EXTENSION_CONNECT_SENT` be set to `false` here?

Reply:
I don't think that's necessary. If we destroy the streams, no messages can trigger 
[extensionStreamMessageListener](https://github.com/MetaMask/metamask-extension/blob/194b1558e65c0ac65b370adb6bc419e4c705de8b/app/scripts/streams/provider-stream.ts#L312-L333) to fire `METAMASK_EXTENSION_CONNECT_CAN_RETRY` in the first place, but at the same time that means it shouldn't hurt to do so

Reply:
I'll add it